sympy/core/relational.py

```diff
@@ -465,23 +467,26 @@ def _eval_simplify(self, **kwargs):
                         constant = c[-1]
                         c[-1] = 0
                         scale = gcd(c)
-                        c = [ctmp / scale for ctmp in c]
-                        r = r.func(Poly.from_list(c, x).as_expr(), -constant / scale)
+                        if scale.is_zero is False:  # avoid dividing by symbolic expressions
+                            c = [ctmp / scale for ctmp in c]
+                            r = r.func(Poly.from_list(c, x).as_expr(), -constant / scale)
                     except PolynomialError:
                         pass
             elif len(free) >= 2:
                 try:
                     from sympy.solvers.solveset import linear_coeffs
                     from sympy.polys.polytools import gcd
+                    from typing import Optional
                     free = list(ordered(free))
                     dif = r.lhs - r.rhs
                     m = linear_coeffs(dif, *free)
                     constant = m[-1]
                     del m[-1]
                     scale = gcd(m)
-                    m = [mtmp / scale for mtmp in m]
-                    nzm = list(filter(lambda f: f[0] != 0, list(zip(m, free))))
-                    if scale.is_zero is False:
+                    scalez: Optional[bool] = scale.is_zero
+                    if scalesz is False:
```

Comment:
```suggestion
                    if scalez is False:
```
You may have written the wrong variable name here
This is warned by `flake8` and doesn't pass the tests.